File: WorldFeed/Server/src/Services/Identity/WorldFeed.Identity.API/Auth/IAuthController.cs
namespace WorldFeed.Identity.API.Auth
{
    using System.Threading.Tasks;

    using WorldFeed.Common.Public.Parameters.Auth;
    using WorldFeed.Identity.API.Application.Requesters;
    using WorldFeed.Identity.API.Application.Web;
    using WorldFeed.Identity.API.DTO;
    using WorldFeed.Identity.API.Models.Authentication;

    public interface IAuthController
    {
        Task<ITwitterResult<CreateTokenResponseDTO>> CreateBearerTokenAsync(ICreateBearerTokenParameters parameters, ITwitterRequest request);

        Task<ITwitterResult<IAuthenticationRequest>> RequestAuthUrlAsync(IRequestAuthUrlParameters parameters, ITwitterRequest request);

        Task<ITwitterResult<ITwitterCredentials>> RequestCredentialsAsync(IRequestCredentialsParameters parameters, ITwitterRequest request);

        Task<ITwitterResult<InvalidateTokenResponse>> InvalidateAccessTokenAsync(IInvalidateAccessTokenParameters parameters, ITwitterRequest request);

        Task<ITwitterResult<InvalidateTokenResponse>> InvalidateBearerTokenAsync(IInvalidateBearerTokenParameters parameters, ITwitterRequest request);
    }
}

File: WorldFeed/Client/src/app/core/Core/Controllers/IAuthController.ts
import {ITwitterRequest} from "../../Public/Models/Interfaces/ITwitterRequest";
import {ITwitterResult} from "../Web/TwitterResult";
import {ICreateBearerTokenParameters} from "../../Public/Parameters/Auth/CreateBearerTokenParameters";
import {IRequestAuthUrlParameters} from "../../Public/Parameters/Auth/IRequestAuthUrlParameters";
import {IRequestCredentialsParameters} from "../../Public/Parameters/Auth/RequestCredentialsParameters";
import {IInvalidateAccessTokenParameters} from "../../Public/Parameters/Auth/InvalidateAccessTokenParameters";
import {IInvalidateBearerTokenParameters} from "../../Public/Parameters/Auth/InvalidateBearerTokenParameters";
import {InvalidateTokenResponse} from "../../Public/Models/Authentication/InvalidateTokenResponse";
import {ITwitterCredentials} from "../../Public/Models/Authentication/TwitterCredentials";
import {IAuthenticationRequest} from "../../Public/Models/Authentication/IAuthenticationRequest";
import {CreateTokenResponseDTO} from "../DTO/CreateTokenResponseDTO";

export interface IAuthController {
  createBearerTokenAsync(parameters: ICreateBearerTokenParameters, request: ITwitterRequest): Promise<ITwitterResult<CreateTokenResponseDTO>>;

  requestAuthUrlAsync(parameters: IRequestAuthUrlParameters, request: ITwitterRequest): Promise<ITwitterResult<IAuthenticationRequest>>;

  requestCredentialsAsync(parameters: IRequestCredentialsParameters, request: ITwitterRequest): Promise<ITwitterResult<ITwitterCredentials>>;

  invalidateAccessTokenAsync(parameters: IInvalidateAccessTokenParameters, request: ITwitterRequest): Promise<ITwitterResult<InvalidateTokenResponse>>;

  invalidateBearerTokenAsync(parameters: IInvalidateBearerTokenParameters, request: ITwitterRequest): Promise<ITwitterResult<InvalidateTokenResponse>>;
}
